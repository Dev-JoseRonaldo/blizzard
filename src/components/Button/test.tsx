import { render, screen } from '@testing-library/react'

import { Button } from '.'

const { getByRole } = screen

describe('<Button />', () => {
  it('should render by default', () => {
    render(<Button>Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('py-3.5 px-8')
    expect(component).toHaveClass('bg-primary hover:bg-blue-600')
  })

  it('should render with custom class', () => {
    render(<Button className="uppercase">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('uppercase')
  })

  it('should render with correct sizes', () => {
    const { rerender } = render(<Button size="small">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('w-[7.2rem] h-[2.56rem] text-sm font-medium')

    rerender(<Button size="full">Children</Button>)

    expect(component).toHaveClass('py-3.5 w-full')
  })

  it('should render with correct variants', () => {
    const { rerender } = render(<Button variant="secondary">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass(
      'hover:text-black-800 hover:bg-white border-white border-2',
    )

    rerender(<Button variant="tertiary">Children</Button>)

    expect(component).toHaveClass('gap-2.5 py-0 px-0 hover:text-primaryLight')
  })

  it('should match snapshot', () => {
    const { container } = render(<Button>Children</Button>)

    expect(container).toMatchSnapshot()
  })
})
