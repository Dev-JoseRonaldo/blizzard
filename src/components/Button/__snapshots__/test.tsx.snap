// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <button
    class="flex items-center justify-center rounded-[3px] text-sans text-white text-base font-semibold transition transition-color duration-300  gap-[0.4rem] py-3.5 px-8 bg-primary hover:bg-blue-600 "
  >
    Children
  </button>
</div>
`;
